server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -621,6 +622,7 @@ public void onResponse(RepositoryData repositoryData) {
                                     return;
                                 }
                                 listener.onResponse(snapshotInfo);
+                                cleanOrphanTimestamp(repositoryName, repositoryData);
```

Comment:
As listener.onResponse is already called, this clean-up will not impact the snapshot creation time, right?